services/src/main/java/org/keycloak/services/filters/CachingGZIPEncodingInterceptor.java

```diff
@@ -0,0 +1,90 @@
+package org.keycloak.services.filters;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.output.TeeOutputStream;
+import org.jboss.resteasy.plugins.interceptors.encoding.GZIPEncodingInterceptor;
+import org.jboss.resteasy.resteasy_jaxrs.i18n.LogMessages;
+import org.keycloak.common.Profile;
+
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.Provider;
+import javax.ws.rs.ext.WriterInterceptorContext;
+import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@Provider
+@Priority(Priorities.ENTITY_CODER)
+public class CachingGZIPEncodingInterceptor extends GZIPEncodingInterceptor {
+
+    private final Path tmpDir;
+    @Context
+    private UriInfo uriInfo;
+
+    public CachingGZIPEncodingInterceptor() {
+        try {
+            this.tmpDir = Files.createTempDirectory("gzip-cache");
+        } catch (IOException e) {
+            throw new RuntimeException("could not create temp gzip cache directory");
+        }
+    }
+
+    public static class CacheCommittedGZIPOutputStream extends CommittedGZIPOutputStream {
+        protected CacheCommittedGZIPOutputStream(final OutputStream delegate) {
+            super(delegate, null);
+        }
+    }
+
+    @Override
+    public void aroundWriteTo(WriterInterceptorContext context) throws IOException, WebApplicationException {
+        LogMessages.LOGGER.debugf("Interceptor : %s,  Method : aroundWriteTo", getClass().getName());
+
+        Object encoding = context.getHeaders().getFirst(HttpHeaders.CONTENT_ENCODING);
```

Comment:
Doesn't client/browser use "accept-encoding" header to tell the server it can accept gzip?

Reply:
Yep that is handled by adding the `@GZip` annotation